File: exercise/ts/day14/src/children/childMapper.ts
import {X5T78} from "./db2/x5T78";
import {Address, Child, Gender} from "./dtos/child";

export class ChildMapper {
    public static toDto(db2Child: X5T78): Child {
        const child = new Child();
        child.Id = db2Child.Id;
        child.FirstName = db2Child.N_1;
        child.MiddleName = db2Child.N_2;
        child.LastName = db2Child.N_3;
        child.BirthCity = db2Child.CityOfBirth__pc;
        child.BirthDate = db2Child.Person_BD;
        child.Gender = db2Child.Salutation === 'Girl' ? Gender.Girl : Gender.Boy;
        child.Address = new Address();
        child.Address.Number = db2Child.ST_Num;
        child.Address.Street = db2Child.ST____Name;
        child.Address.City = db2Child.ST_C;
        child.Address.CountryId = parseInt(db2Child.ST_CID, 10);

        return child;
    }
}

File: exercise/scala/day14/src/main/scala/children/ChildMapper.scala
package children

import children.db2.X5T78
import children.dtos.{Address, Child, Gender}

import java.util.UUID

object ChildMapper {
  def toDto(child: X5T78): Child = {
    Child(
      id = child.id.map(UUID.fromString).getOrElse(UUID.randomUUID()),
      firstName = child.n1,
      middleName = child.n2,
      lastName = child.n3,
      birthCity = child.cityOfBirthPc,
      birthDate = child.personBd,
      gender = child.salutation match {
        case Some("Girl") => Gender.Girl
        case Some("Boy") => Gender.Boy
        case _ => throw new IllegalArgumentException("Unknown gender")
      },
      address = for {
        number <- child.stNum
        street <- child.stName
        city <- child.stC
        countryId <- child.stCid.map(id => id.toInt)
      } yield Address(number, street, city, countryId)
    )
  }
}